client/task-list/tasks/payments/RemotePayments/index.js

```diff
@@ -19,27 +18,22 @@ import { sift } from '../../../../utils';
 
 export const RemotePayments = ( { query } ) => {
 	const { updateOptions } = useDispatch( OPTIONS_STORE_NAME );
-	const { getOption } = useSelect( ( select ) => {
+	const {
+		getOption,
+		getPaymentMethodRecommendations,
+		isFetching,
+	} = useSelect( ( select ) => {
 		return {
 			getOption: select( OPTIONS_STORE_NAME ).getOption,
+			getPaymentMethodRecommendations: select( ONBOARDING_STORE_NAME )
+				.getPaymentMethodRecommendations,
+			isFetching: select( ONBOARDING_STORE_NAME ).hasFinishedResolution(
+				'getPaymentMethodRecommendations'
+			),
 		};
 	} );
 
-	const [ methods, setMethods ] = useState( [] );
-	const [ isFetching, setIsFetching ] = useState( true );
-
-	useEffect( () => {
-		apiFetch( {
-			path: '/wc-admin/onboarding/payments',
-		} )
-			.then( ( results ) => {
-				setMethods( results );
-				setIsFetching( false );
-			} )
-			.catch( () => {
-				setIsFetching( false );
-			} );
-	}, [] );
+	const methods = getPaymentMethodRecommendations();
```

Comment:
Thats a worthwhile conversation to have. If Core uses `gateway`, I like `paymentGatewayRecommendations` for consistency. Also, when we make a decision here, it seems revisiting the name of the endpoint introduced in https://github.com/woocommerce/woocommerce-admin/pull/6830 makes sense too. In this case it could be `/wp-json/wc-admin/onboarding/payment-gateway-recommendations`.

Reply:
I do think `paymentGatewayRecommendations` sounds like the best choice since core uses `gateway`. If a decision can be made before this PR is approved and merged, I'll make the changes. A question, would we want to also rename things like `enabledMethods` to `enabledGateways`? I think it can make sense to rename those too but it'll be quite an undertaking.

Reply:
Great thoughts, @ilyasfoo and @psealock.  I think we can merge this one first and do a full audit of all the semantics in a follow-up.